src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysis.java

```diff
@@ -342,4 +386,22 @@ private static boolean violationWeakenedOrEquivalent(LimitViolation violation1,
     List<LfContingency> createContingencies(List<PropagatedContingency> propagatedContingencies, LfNetwork network) {
         return LfContingency.createContingencies(propagatedContingencies, network, network.createDecrementalConnectivity(connectivityProvider), true);
     }
+
+    private void addMonitorInfos(LfNetwork lfNetwork, StateMonitor monitor, Consumer<BranchResult> branchResultConsumer,
+                                 Consumer<BusResults> busResultsConsumer) {
+        monitor.getBranchIds().forEach(branchId -> branchResultConsumer.accept(createBranchResult(lfNetwork.getBranchById(branchId))));
+        lfNetwork.getBuses().stream().filter(lfBus -> monitor.getVoltageLevelIds().contains(lfBus.getVoltageLevelId())).forEach(lfBus ->
+            busResultsConsumer.accept(createBusResult(lfBus, lfBus.getVoltageLevelId())));
+    }
+
+    private BranchResult createBranchResult(LfBranch branch) {
+        double currentScale = PerUnit.SB / branch.getBus1().getNominalV();
```

Comment:
Here I am not sure that it is robust to get the nominal voltage through the bus. I think you should try to put this method in the LfBranch directly and use attributes.

Reply:
Add a unit test with open branches, dangline line and three windings transformer.